### PR TITLE
Optimized number of lines fetching in log file initialisation

### PR DESCRIPTION
**Pull request by seuf**
I've optimized the way of counting the number of lines in the existing log file.
thanks stack overflow 😆  : https://stackoverflow.com/questions/24562942/golang-how-do-i-determine-the-number-of-lines-in-a-file-efficiently#24563853

Related Issue : https://github.com/grafana/grafana/issues/9948

### CODE REVIEW

**Comment by torkelo** [comment]
great! thanks for contributing!

**Comment by bergquist** [comment]
@seuf awesome! any chance you can add a test for this? Just reading a file and making sure the function returns the correct amount of rows would be enough.

**Review by bergquist — CHANGES_REQUESTED** [review]
@seuf awesome! any chance you can add a test for this? Just reading a file and making sure the function returns the correct amount of rows would be enough.

**Comment by seuf** [comment]
unit tests added :)

File: pkg/log/file_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	return nil
+}
+
+func TestLogFile(t *testing.T) {
+
+	Convey("When logging to file", t, func() {
+		fileLogWrite := NewFileWriter()
+		So(fileLogWrite, ShouldNotBeNil)
+
+		fileLogWrite.Filename = "../../data/log/grafana.log"
```
Comment (bergquist) [inline pkg/log/file_test.go]: using the standard logfile will fail when running the test locally if my log file have content. Just use a local file in this directory. The path is not important in this case.

File: pkg/log/file_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	if err != nil {
+		return err
+	}
+	w.docheck(n)
+	return nil
+}
+
+func TestLogFile(t *testing.T) {
+
+	Convey("When logging to file", t, func() {
```
Comment (bergquist) [inline pkg/log/file_test.go]: Please remove the test log file after all the test are complete.

**Review by bergquist — CHANGES_REQUESTED** [review]

File: pkg/log/file_test.go (hunk 2 above)
Comment (seuf) [reply pkg/log/file_test.go]: ok.. log file is removed at the end of the tests.

Comment (bergquist) [reply pkg/log/file_test.go]: me doing code reviews 
image

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Thank you for contributing!

**Pull request merged**